internal/coreinternal/scraperinttest/scraperint.go

```diff
@@ -113,6 +121,31 @@ func (it *IntegrationTest) Run(t *testing.T) {
 		it.compareTimeout, it.compareTimeout/20)
 }
 
+func (it *IntegrationTest) createNetwork(t *testing.T) testcontainers.Network {
+	var errs error
+	defer func() {
+		if t.Failed() && errs != nil {
+			t.Errorf("create network: %v", errs)
```

Comment:
Maybe just put this error message as another optional argument to `require.Eventually` instead of this defer block?

Reply:
You're right - this pattern is pointless when used like this.

It took me a minute to recall why I was using this pattern in the first place, but the original idea was to have a dynamic error message, which I think can be more helpful in understanding what went wrong. But I wasn't actually using it that way in this case.